src/component.ts

```diff
@@ -49,11 +63,24 @@ function makeComponent(render: RenderFunction): Creator {
     const BaseElement = (options || baseElementOrOptions as Options<P> || {}).baseElement || HTMLElement;
     const {observedAttributes = [], useShadowDOM = true, shadowRootInit = {}} = options || baseElementOrOptions as Options<P> || {};
 
+    const _observedAttributes =
```

Comment:
This isn't exactly the best name for this. It might make more sense to call it `observedAttributes` and remove the one above and replace all of its instances with `options.observedAttributes`. That should remove any confusion as to which "observedAttributes" is the real one.

```js
const observedAttributes = renderer.observedAttributes || options.observedAttributes || [];
```